tests/test_run_metrics.py

```diff
@@ -0,0 +1,64 @@
+"""
+Tests for the run metrics script
+"""
+
+from locomoset.metrics.run import (
+    compute_metric,
+    nest_string_in_list,
+    parameter_sweep_dicts,
+)
+
+
+def test_parameter_sweep():
+    """
+    Test the parameter sweep config dicts generator function
+    """
+    test_dict = {
```

Comment:
in the example you've given the itertools function would treat "name_1" as an iterable and sweep over the characters of the string for each parameter set, hence requiring any string be wrapped in a list so it gets picked up as a single element of an iterable rather than an iterable itself. 

The "par_3" value would remain constant in this case.

Reply:
@eddableheath can respond here but my understanding is it already works as written for other single, non-iterable things, the problem is just the edge-case that a string is treated as an iterable of characters rather than a single entity.

Reply:
@jack89roberts I think it should give an error `TypeError: 'int' object is not iterable` if you try the test with the `"par_3"` param included.
@eddableheath I think the logic should work? 
```
>>> from typing import Iterable
>>> maybe_box = lambda x: [x] if not isinstance(x, Iterable) or isinstance(x, str) else x
>>> maybe_box("hello")
['hello']
>>> maybe_box(["hello"])
['hello']
>>> maybe_box(1)
[1]
>>> maybe_box([1])
[1]
```

Reply:
the above fn would replace `nest_string_in_list` though perhaps not as a lambda!

Reply:
yes, you're completely correct, will update the function and the tests

Reply:
fixed with this commit [14e3317](https://github.com/alan-turing-institute/ARC-LoCoMoSeT/pull/15/commits/14e331740b402c7ff268a55261448580312820a2)